src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -224,11 +225,17 @@ contract LandBaseTokenV3 is ERC721BaseTokenV2 {
         uint256 x,
         uint256 y
     ) public view returns (bool) {
-        require(x % size == 0 && y % size == 0, "Invalid coordinates");
-        require(x <= GRID_SIZE - size && y <= GRID_SIZE - size, "Out of bounds");
+        _isValidCoordinates(x, y, size);
         return _ownerOfQuad(size, x, y) != address(0);
     }
 
+    function _isValidCoordinates(uint256 x, uint256 y, uint256 size) internal pure {
+        require(x % size == 0, "Invalid x coordinate");
+        require(y % size == 0, "Invalid y coordinate");
+        require(x <= GRID_SIZE - size, "x out of bounds");
+        require(y <= GRID_SIZE - size, "y out of bounds");
+    }
```

Comment:
I think you should redo H-01 based on this function and add the require on the sizes here

Reply:
also, can we name it the same way than in 0.8 ? 
_isValidQuad